var-naming: don't use leading k in Go names, vague and unclear failure description

Seeing this linter failure message.
```
[2023-08-11T05:40:28.300Z] cmd/something/something.go:15:3: var-naming: don't use leading k in Go names; var kPath should be path (revive)
```

The issue with the failure message is that it effectively describes 'what' is wrong, but is vague on explaining 'why' it's wrong. Initially, after some research, I thought the problem might be related to the guideline in idiomatic Go which advises against [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation). However, upon closer inspection of the code, it's evident that the check is solely for variable names prefixed with a lowercase `k`.

https://github.com/mgechev/revive/blob/b4fc3db69235d2ce2a394bbd84f9f8f0ff937e7d/rule/var-naming.go#L142-L150

It would be helpful to either annotate the code with comments or refine the failure message to clarify the 'why' behind this check within the `var-naming` rule.

As an alternative, perhaps the check should be designed to identify Hungarian notation. Or, maybe it's worth considering the removal of this check altogether?

--- a/rule/var-naming.go
+++ b/rule/var-naming.go
@@ -141,15 +141,6 @@ func (w *lintNames) check(id *ast.Ident, thing string) {
 		})
 		return
 	}
-	if len(id.Name) > 2 && id.Name[0] == 'k' && id.Name[1] >= 'A' && id.Name[1] <= 'Z' {
-		should := string(id.Name[1]+'a'-'A') + id.Name[2:]
-		w.onFailure(lint.Failure{
-			Failure:    fmt.Sprintf("don't use leading k in Go names; %s %s should be %s", thing, id.Name, should),
-			Confidence: 0.8,
-			Node:       id,
-			Category:   "naming",
-		})
-	}
 
 	should := lint.Name(id.Name, w.whitelist, w.blacklist)
 	if id.Name == should {

--- a/testdata/golint/var-naming.go
+++ b/testdata/golint/var-naming.go
@@ -57,8 +57,7 @@ func f_it() { // MATCH /don't use underscores in Go names; func f_it should be f
 
 // Common styles in other languages that don't belong in Go.
 const (
-	CPP_CONST   = 1 // MATCH /don't use ALL_CAPS in Go names; use CamelCase/
-	kLeadingKay = 2 // MATCH /don't use leading k in Go names; const kLeadingKay should be leadingKay/
+	CPP_CONST = 1 // MATCH /don't use ALL_CAPS in Go names; use CamelCase/
 
 	HTML  = 3 // okay; no underscore
 	X509B = 4 // ditto

